Tests/FoundationInternationalizationTests/Formatting/NumberParseStrategyTests.swift

```diff
@@ -192,22 +192,68 @@ final class NumberParseStrategyTests : XCTestCase {
     func testNumericBoundsParsing() throws {
         let locale = Locale(identifier: "en_US")
         do {
+            let format: IntegerFormatStyle<Int8> = .init(locale: locale)
+            let parseStrategy = IntegerParseStrategy(format: format, lenient: true)
+            XCTAssertEqual(try parseStrategy.parse(Int8.min.formatted(format)), Int8.min)
+            XCTAssertEqual(try parseStrategy.parse(Int8.max.formatted(format)), Int8.max)
+            XCTAssertThrowsError(try parseStrategy.parse("-129"))
+            XCTAssertThrowsError(try parseStrategy.parse("128"))
+        }
+        
+        do {
+            let format: IntegerFormatStyle<Int64> = .init(locale: locale)
+            let parseStrategy = IntegerParseStrategy(format: format, lenient: true)
+            XCTAssertEqual(try parseStrategy.parse(Int64.min.formatted(format)), Int64.min)
+            XCTAssertEqual(try parseStrategy.parse(Int64.max.formatted(format)), Int64.max)
+            XCTAssertThrowsError(try parseStrategy.parse("-9223372036854775809"))
+            XCTAssertThrowsError(try parseStrategy.parse("9223372036854775808"))
+        }
+        
+        do {
+            let format: IntegerFormatStyle<UInt8> = .init(locale: locale)
+            let parseStrategy = IntegerParseStrategy(format: format, lenient: true)
+            XCTAssertEqual(try parseStrategy.parse(UInt8.min.formatted(format)), UInt8.min)
+            XCTAssertEqual(try parseStrategy.parse(UInt8.max.formatted(format)), UInt8.max)
+            XCTAssertThrowsError(try parseStrategy.parse("-1"))
+            XCTAssertThrowsError(try parseStrategy.parse("256"))
+        }
+        
+        do {
+            // TODO: Parse integers greater than Int64
             let format: IntegerFormatStyle<UInt64> = .init(locale: locale)
             let parseStrategy = IntegerParseStrategy(format: format, lenient: true)
-            XCTAssertEqual(try parseStrategy.parse(0.formatted(format)), 0)
-            let aboveInt64Max = UInt64(Int64.max) + 1
-            XCTAssertEqual(try parseStrategy.parse(aboveInt64Max.formatted(format)), aboveInt64Max)
+            XCTAssertEqual(try parseStrategy.parse(UInt64.min.formatted(format)), UInt64.min)
+            //XCTAssertEqual(try parseStrategy.parse(UInt64.max.formatted(format)), UInt64.max)
```

Comment:
What's the issue with this? Does this currently crash?

Anyways if we are going to leave the tests commented out, a comment with an issue number would be useful

Reply:
Values greater than Int64.max go through Double before this patch, but such values are considered out-of-bounds after this patch. 

Int64.max + 1 is an exact Double, so the current fallback path formats it correctly. The problem is that the subsequent 2047 values are not Double(s) so they don't get formatted as expected. I saw no tests covering integers between Double(s), so I've assumed the current Int64.max + x behavior is an oversight. I could assert that it throws an error, perhaps with some comments about ICU limitations, but that seems like a temporary solution.

I suppose this matters more when decoding JSON. I thought about amending the JSON stuff first since there's a code comment about losslessness there, but it is much easier to discuss the formatting case since it has a straightforward test suite.

Reply:
Thanks yeah that's reasonable. Can we add the test back and assert that it would now throw? Same applies to other commented-out tests below